translator/translate/otel/pipeline/applicationsignals/translator.go

```diff
@@ -52,12 +52,23 @@ func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators
 	}
 
 	translators := &common.ComponentTranslators{
-		Receivers:  common.NewTranslatorMap(otlp.NewTranslator(common.WithName(common.AppSignals), otlp.WithSignal(t.signal))),
+		Receivers:  common.NewTranslatorMap[component.Config, component.ID](),
 		Processors: common.NewTranslatorMap[component.Config, component.ID](),
 		Exporters:  common.NewTranslatorMap[component.Config, component.ID](),
 		Extensions: common.NewTranslatorMap[component.Config, component.ID](),
 	}
 
+	// Add OTLP receivers
+	otlps, err := otlp.ParseOtlpConfig(conf, common.AppSignals, "", t.signal, -1)
```

Comment:
Im wondering if this contract here should still be the same i.e. we just call `otlp.NewTranslator` here and then we just tweak `NewTranslator` to internally have this logic and also invoke `ParseOtlpConfig` internally.

For reference - https://github.com/aws/amazon-cloudwatch-agent/blob/sky333999/otlp/translator/translate/otel/receiver/otlp/translator.go. In this file, my `translate` is the equivalent of your `ParseOtlpConfig`.

Reply:
I think the difference is that `ParseOtlpConfig` can generate multiple OTLP receivers, so would have to be a `NewTranslators` function. I think that would make it clearer what the purpose is. Could just have it return `[]common.ComponentTranslator`.